src/extractor/edge_based_graph_factory.cpp

```diff
@@ -638,6 +595,475 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
+        if (turn_candidates[getRight(0)].instruction == TurnInstruction::UTurn)
+            turn_candidates[getRight(0)].instruction = TurnInstruction::TurnSharpRight;
+    }
+
+    const auto keepStraight = [](double angle)
+    {
+        return std::abs(angle - 180) < 5;
+    };
+
+    if (turn_candidates.size() > 1)
+    {
+        for (std::size_t turn_index = 0; turn_index < turn_candidates.size(); ++turn_index)
+        {
+            auto &turn = turn_candidates[turn_index];
+            if (turn.instruction > TurnInstruction::TurnSlightLeft ||
+                turn.instruction == TurnInstruction::UTurn)
+                continue;
+            auto &left = turn_candidates[getLeft(turn_index)];
+            if (turn.angle == left.angle)
+            {
+                util::SimpleLogger().Write()
+                    << "[warning] conflicting turn angles, identical road duplicated? "
+                    << m_node_info_list[m_node_based_graph->GetTarget(via_eid)].lat << " "
+                    << m_node_info_list[m_node_based_graph->GetTarget(via_eid)].lon << std::endl;
+            }
+            if (isConflict(turn.instruction, left.instruction))
+            {
+                // begin of a conflicting region
+                std::size_t conflict_begin = turn_index;
+                std::size_t conflict_end = getLeft(turn_index);
+                std::size_t conflict_size = 2;
+                while (isConflict(turn_candidates[getLeft(conflict_end)].instruction,
+                                  turn.instruction) &&
+                       conflict_size < turn_candidates.size())
+                {
+                    conflict_end = getLeft(conflict_end);
+                    ++conflict_size;
+                }
+
+                turn_index =
+                    (conflict_end < conflict_begin) ? turn_candidates.size() : conflict_end;
+
+                if (conflict_size > 3)
+                {
+                    // check if some turns are invalid to find out about good handling
+                }
+
+                auto &instruction_left_of_end = turn_candidates[getLeft(conflict_end)].instruction;
+                auto &instruction_right_of_begin =
+                    turn_candidates[getRight(conflict_begin)].instruction;
+                auto &candidate_at_end = turn_candidates[conflict_end];
+                auto &candidate_at_begin = turn_candidates[conflict_begin];
+                if (conflict_size == 2)
+                {
+                    if (turn.instruction == TurnInstruction::GoStraight)
+                    {
+                        if (instruction_left_of_end != TurnInstruction::TurnSlightLeft &&
+                            instruction_right_of_begin != TurnInstruction::TurnSlightRight)
+                        {
+                            std::int32_t resolved_count = 0;
+                            if (!keepStraight(candidate_at_end.angle) &&
```

Comment:
Why the inverted logic here? `if (keepStraight() || resolve())` and switch branches.